### PR TITLE
Add new Resources tab to display book recommendations

### PR DESCRIPTION
**Pull request by lilyndk**
## Description
Resources Tab has been added to the mentor profile.
Two scenarios have been implemented:
1) If mentor has book recommendations -> display recommendations in a list with book title and comment (optional).
2) If mentor has no recommendations yet -> Display "This mentor doesn't have resource recommendations yet.".

Data: books.yml which is mapped with mentors.yml using mentor's index.

## Change Type
- [ ] Bug Fix
- [X] New Feature
- [ ] Code Refactor
- [ ] Documentation
- [ ] Other

## Related Issue
N/A

## Motivation and Context
Existing Trello task: https://trello.com/c/qqtK2Zbu

## Screenshots
After implementation:
Screenshot 2024-03-01 at 14 47 13
Screenshot 2024-03-01 at 14 49 24

Before implementation (no Resources tab):
Screenshot 2024-02-29 at 23 11 19

## Pull request checklist

Please check if your PR fulfills the following requirements:
- [X] I checked and followed the [contributor guide](CONTRIBUTING.md) 
- [X] I have tested my changes locally.
- [X] I have added a screenshot from the website after I tested it locally

### CODE REVIEW

File: _data/books.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    - title: "Clean Code: A Handbook of Agile Software Craftsmanship, by Robert C. Martin"
+      comments: "Great for learning how to level up the quality of code base."
+    - title: "Designing Data-Intensive Applications: The Big Ideas Behind Reliable, Scalable, and Maintainable Systems, by Martin Kleppmann"
+      comments: "So many great concepts in this book!"
+    - title: "Domain-Driven Design: Tackling Complexity in the Heart of Software, by Eric Evans"
+      comments: "Discover concepts for creating well-designed software."
+    - title: "Release it!: Design and Deploy Production-Ready Software, by Michael T. Nygard"
+      comments: "Learn about stability patterns in software."
+    - title: "No Rules Rules: Netflix and the Culture of Reinvention, by Reed Hastings and Erin Meyer"
+      comments: "Amazing book around team culture."
```
Comment (dricazenck) [inline _data/books.yml:23]: It is not necessary to add comma, and when is empty you don't need to add.

Comment (lilyndk) [reply _data/books.yml:23]: gotcha, I'll fix it :)

Comment (lilyndk) [reply _data/books.yml:23]: @dricazenck fixed in latest commit

File: _data/books.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+- index: 1 # mentor id 1
+  books:
+    - title: "Thinking, Fast and Slow by Daniel Kahneman"
+      comments:
+    - title: "Never Split the Difference: Negotiating as if Your Life Depended on It  by Chris Voss and Tahl Raz"
+      comments:
+    - title: "The Culture Map - Breaking Through the Invisible Boundaries of Global Business by Erin Meyer"
```
Comment (dricazenck) [inline _data/books.yml:7]:
```suggestion
    - title: |
    The Culture Map - Breaking Through the Invisible Boundaries of Global Business by Erin Meyer
```

File: _data/books.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+- index: 1 # mentor id 1
+  books:
+    - title: "Thinking, Fast and Slow by Daniel Kahneman"
```
Comment (dricazenck) [inline _data/books.yml:3]:
```suggestion
    - title: Thinking, Fast and Slow by Daniel Kahneman
```

File: _data/books.yml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+      comments:
```
Comment (dricazenck) [inline _data/books.yml:4]:
```suggestion
     
```

File: _data/books.yml [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    - title: "Never Split the Difference: Negotiating as if Your Life Depended on It  by Chris Voss and Tahl Raz"
+      comments:
+    - title: "The Culture Map - Breaking Through the Invisible Boundaries of Global Business by Erin Meyer"
+      comments:
+    - title: "The Pragmatic Programmer: From Journeyman to Master by Andy Hunt and Dave Thomas"
+      comments:
+
+- index: 7 # mentor id 7
+  books:
+    - title: "Clean Code: A Handbook of Agile Software Craftsmanship by Robert C. Martin"
```
Comment (dricazenck) [inline _data/books.yml:14]: What I mentioned was that if you don't have any characters to break the syntax you can just add the text. 

In this case test you have this : you can solve with |
```suggestion
    - title: |
    Clean Code: A Handbook of Agile Software Craftsmanship by Robert C. Martin"
```

File: mentors.html [hunk 6]
```diff
@@ -166,6 +169,27 @@ <h5 class="position">{{mentor.position}}</h5>
...
+                                </div>
+                            {% endif %}
+                            {% for book in books %}
+                                <div id="mentor-resources-{{mentor.index}}" class="card-resources d-none">
+                                    <h5>Book Recommendations</h5>
+                                    {% assign books = book.books %}
+                                    <ul>
+                                        {% for book in books %}
+                                        <li>
+                                            <h6>{{ book.title }}</h6>
```
Comment (dricazenck) [inline mentors.html:186]: I think here should be a div or it is necessary to add more emphasis into the previous session. They look like the same size now, maybe add also bold in books.

**Review by dricazenck — COMMENTED** [review]
I liked a lot!!! Thanks for the contribution! 

there are only minors Changes, but once you change it should be ready 🥰👏👏

File: mentors.html (hunk 6 above)
Comment (lilyndk) [reply mentors.html:186]: @dricazenck Thanks for the suggestions! Should be updated now in latest commit. I've also updated the screenshots to show the latest styles with the bold book titles. FYI, @nora-weisser said that she will add her book recommendations' comments later on :)

**Review by dricazenck — APPROVED** [review]

**Pull request merged**